src/stdune/user_message.mli

```diff
@@ -0,0 +1,66 @@
+(** A message for the user *)
+
+(** User messages are styled document that can be printed to the
+    console or in the log file. *)
+
+(** Styles that can be used inside messages *)
+module Style : sig
```

Comment:
I don't really understand what the abstract styles mean. It looks like most of them are not used?
If we want to make something future-proof, shouldn't it just make it extensible? For example, agree to never remove or rename a style, but that we can add a style later.

Reply:
They are meant to be symbolic styles that are converted to concrete ones (such as ansi terminal styles) before rendering to a particular output device. I added a bit more doc.

With the old error reporting API, they are passed inside format strings via `"@{<error>"`.